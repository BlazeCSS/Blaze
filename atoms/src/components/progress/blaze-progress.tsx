import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'blaze-progress'
})
export class Progress {

  @Prop() type: string;
  @Prop() percentage: string;
  @Prop() rounded: boolean;
  @Prop() size: string = '';

  render() {
    const typeClass = this.type ? `c-progress__bar--${this.type}` : '';
    const sizeClass = this.size ? `u-${this.size}` : '';
    const roundedClass = this.rounded ? `c-progress--rounded` : '';

    return (
      <div class={`c-progress ${roundedClass} ${sizeClass}`}>
        <div class={`c-progress__bar ${typeClass}`} style={{width:`${this.percentage}%`}}>
          <slot/>
        </div>
      </div>
    );
  }
}
