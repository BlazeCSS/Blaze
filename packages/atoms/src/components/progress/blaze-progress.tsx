import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'blaze-progress'
})
export class Progress {

  @Prop() rounded: boolean;
  @Prop() size: string = '';

  render() {
    const sizeClass = this.size ? `u-${this.size}` : '';
    const roundedClass = this.rounded ? `c-progress--rounded` : '';

    return (
      <div class={`c-progress ${roundedClass} ${sizeClass}`}>
        <slot />
      </div>
    );
  }
}
