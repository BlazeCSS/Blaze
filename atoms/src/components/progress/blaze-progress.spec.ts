import { flush, render } from '@stencil/core/testing';
import { Progress } from './blaze-progress';

describe('Progress', () => {
  it('should build', () => {
    expect(new Progress()).toBeTruthy();
  });

  let element;

  const snapIt = (name, html) => {
    it(name, async () => {
      element = await render({
        components: [Progress],
        html
      });
      await flush(element);

      expect(element).toMatchSnapshot();
    });
  };


  snapIt(
    'renders basic progress bar',
    '<blaze-progress percentage="5">5%</blaze-progress>'
  );

  snapIt(
    'renders size correctly',
    '<blaze-progress size="small" percentage="15">15%</blaze-progress>'
  );

  snapIt(
    'renders type correctly',
    '<blaze-progress size="medium" type="info" percentage="20">20%</blaze-progress>'
  );

  snapIt(
    'renders rounded ends correctly',
    '<blaze-progress size="large" type="success" percentage="25" rounded>25%</blaze-progress>'
  );
});
