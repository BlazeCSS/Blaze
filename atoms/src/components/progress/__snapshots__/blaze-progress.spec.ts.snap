// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders basic progress bar 1`] = `
<blaze-progress
  class="hydrated"
  percentage="5"
>
  <div
    class="c-progress"
  >
    <div
      class="c-progress__bar"
      style="width: 5%;"
    >
      <!---->
      5%
    </div>
  </div>
</blaze-progress>
`;

exports[`Progress renders rounded ends correctly 1`] = `
<blaze-progress
  class="hydrated"
  percentage="25"
  rounded=""
  size="large"
  type="success"
>
  <div
    class="c-progress c-progress--rounded u-large"
  >
    <div
      class="c-progress__bar c-progress__bar--success"
      style="width: 25%;"
    >
      <!---->
      25%
    </div>
  </div>
</blaze-progress>
`;

exports[`Progress renders size correctly 1`] = `
<blaze-progress
  class="hydrated"
  percentage="15"
  size="small"
>
  <div
    class="c-progress u-small"
  >
    <div
      class="c-progress__bar"
      style="width: 15%;"
    >
      <!---->
      15%
    </div>
  </div>
</blaze-progress>
`;

exports[`Progress renders type correctly 1`] = `
<blaze-progress
  class="hydrated"
  percentage="20"
  size="medium"
  type="info"
>
  <div
    class="c-progress u-medium"
  >
    <div
      class="c-progress__bar c-progress__bar--info"
      style="width: 20%;"
    >
      <!---->
      20%
    </div>
  </div>
</blaze-progress>
`;
